modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java

```diff
@@ -693,17 +690,49 @@ private void _addObjectDefinitionResources(
 				).build();
 
 			for (ObjectRelationship objectRelationship : objectRelationships) {
-				objectRelationship =
-					objectRelationshipResource.
-						postObjectDefinitionObjectRelationship(
-							objectDefinitionId, objectRelationship);
+				String objectRelationshipName = objectRelationship.getName();
+
+				objectRelationshipsNames.add(objectRelationshipName);
+
+				com.liferay.object.model.ObjectRelationship
+					oldObjectRelationship =
+						_objectRelationshipLocalService.
+							fetchObjectRelationshipByObjectDefinitionId(
+								objectDefinitionId, objectRelationshipName);
+
+				if (oldObjectRelationship == null) {
+					objectRelationship =
+						objectRelationshipResource.
+							postObjectDefinitionObjectRelationship(
+								objectDefinitionId, objectRelationship);
+
+					if (accountEntryRestrictedObjectRelationshipsNames.contains(
+							objectRelationshipName)) {
+
+						_objectDefinitionLocalService.
+							enableAccountEntryRestricted(
+								_objectRelationshipLocalService.
+									getObjectRelationship(
+										objectRelationship.getId()));
+					}
+				}
+				else {
+					objectRelationshipResource.putObjectRelationship(
+						oldObjectRelationship.getObjectRelationshipId(),
+						objectRelationship);
+				}
+			}
+
+			for (com.liferay.object.model.ObjectRelationship
+					oldObjectRelationship :
+						_objectRelationshipLocalService.getObjectRelationships(
+							objectDefinitionId)) {
 
-				if (accountEntryRestrictedObjectRelationshipsNames.contains(
-						objectRelationship.getName())) {
+				if (!objectRelationshipsNames.contains(
+						oldObjectRelationship.getName())) {
 
-					_objectDefinitionLocalService.enableAccountEntryRestricted(
-						_objectRelationshipLocalService.getObjectRelationship(
-							objectRelationship.getId()));
+					_objectRelationshipLocalService.deleteObjectRelationship(
+						oldObjectRelationship.getObjectRelationshipId());
```

Comment:
Hey @mateussgn,

take a look at this [method](https://github.com/liferay/liferay-portal/blob/e31246835a1478b66c4f18614e9f3ab319cdec7e/portal-kernel/src/com/liferay/portal/kernel/util/SetUtil.java#L230-L249) and [this one](https://github.com/liferay/liferay-portal/blob/e31246835a1478b66c4f18614e9f3ab319cdec7e/portal-kernel/src/com/liferay/portal/kernel/util/SetUtil.java#L283-L311) as well.

I believe we can try to reuse them or at least the intersect one.

the overachievement for these changes would be this, take a look at this image:

![image](https://github.com/liferay-objects/liferay-portal/assets/42913501/85a23cca-4742-42d3-8ad2-89ccb86adc37)

so the goal is to provide a method in SetUtil that retrieves an object or a map with 3 collections, 

```
left -> item 1, item 2 and item 3
intersection -> item 4
right -> item 5 and item 6
```

so we'd able to add the left collection, update the intersection collection and delete the right collection, and we would be able to use this in other places like state manager for example because there we probably are duplicating this logic.

Thanks!

Reply:
Perfect! Great suggestion!
I didn't know those methods already exits.
I'll try to reuse them.